app/assets/javascripts/toc.js

```diff
@@ -42,6 +38,14 @@
           setCurrentLink(`#${heading.id}`);
         }
       });
+    };
+
+    window.addEventListener("scroll", handleScroll);
+
+    tocLinkNodes.forEach((link) => {
+      link.addEventListener("click", (e) => {
+        setTimeout(() => setCurrentLink(e.target.hash), 25);
```

Comment:
What's the thinking behind adding the timeout here?

Reply:
So that it highlights link that you've clicked as the active TOC item, rather than the heading in the viewport.

What's happening is both the scroll position of heading and the clicked TOC item are competing, and the scroll event always trump.

I tried a bunch of different ways and this was the best I could come up with to ensure the click comes after the scroll position with setTimeout.